### PR TITLE
clarify local and input variables

### PR DESCRIPTION
**Pull request by alanszlosek**
Clarify details for local and input variables:

* How they are evaluated and overridden
* List what a local variable expression can contain
* Fix Local variables page title
* Replace "local values" term

### CODE REVIEW

File: website/content/docs/templates/hcl_templates/locals.mdx [hunk 1]
```diff
@@ -1,35 +1,34 @@
 ---
-page_title: Local Values - HCL Configuration Language
+page_title: Local Variables - HCL Configuration Language
 description: >-
-  Local values assign a name to an expression that can then be used multiple
+  Local variables assign a name to an expression that can then be used multiple
```
Comment (rkoron007) [inline website/content/docs/templates/hcl_templates/locals.mdx:4]: I'm not sure if descriptions should be active or passive voice so feel free to disregard if this is wrong:
```suggestion
  Local variables assign a name to an expression that you can use multiple
```

File: website/content/docs/templates/hcl_templates/locals.mdx [hunk 2]
```diff
@@ -1,35 +1,34 @@
...
 `@include 'from-1.5/beta-hcl2-note.mdx'`
 
 There are two kinds of variables in HCL Packer templates: Input variables,
 sometimes simply called "variables", and Local variables, also known as
-"locals". Input variables may have defaults, but those defaults can
-be overridden from the command line or special variable files. Local variables
-can be thought of as constants, and are not able to be overridden at runtime.
+"locals". Input variables may have defaults, but those defaults can be
+overridden using command line options, environment variables, or variable
+definitions files. Local variables cannot be overridden.
```
Comment (rkoron007) [inline website/content/docs/templates/hcl_templates/locals.mdx:16]: I believe we avoid contractions:
```suggestion
definitions files. Local variables can not be overridden.
```

File: website/content/docs/templates/hcl_templates/locals.mdx [hunk 3]
```diff
@@ -1,35 +1,34 @@
...
 
 This page is about local variables. To learn about input variables, see the
 [input variables](/packer/docs/templates/hcl_templates/variables) page.
 
-Local values assign a name to an expression, that can then be used multiple
-times within a folder.
-
-If [variables](/packer/docs/templates/hcl_templates/variables) are analogous to function arguments then
-_local values_ are comparable to a function's local variables.
+Local variables assign a name to an expression, that can then be used multiple
```
Comment (rkoron007) [inline website/content/docs/templates/hcl_templates/locals.mdx:21]:
```suggestion
Local variables assign a name to an expression, which you can use multiple
```

**Review by rkoron007 — APPROVED** [review]
Some tiny suggestions for style, but otherwise looks great to me!

File: website/content/docs/templates/hcl_templates/locals.mdx [hunk 4]
```diff
@@ -1,35 +1,34 @@
 ---
-page_title: Local Values - HCL Configuration Language
+page_title: Local Variables - HCL Configuration Language
```
Comment (nywilken) [inline website/content/docs/templates/hcl_templates/locals.mdx:2]: I wonder if we should update this document to be consistent with the Terraform [local values documentation](https://developer.hashicorp.com/terraform/language/values/locals). What do you think?

File: website/content/docs/templates/hcl_templates/locals.mdx [hunk 5]
```diff
@@ -1,35 +1,34 @@
...
+reference input variables, other local variables, data sources, and HCL
+functions.
 
 Input variable and local variable usage are introduced in the [_Variables
 Guide_](/packer/guides/hcl/variables).
 
 ## Examples
 
-Local values are defined in `local` or `locals` blocks:
+Local variables are defined in `local` or `locals` blocks:
```
Comment (nywilken) [inline website/content/docs/templates/hcl_templates/locals.mdx:31]:
```suggestion
Local variables are defined in a `local` or `locals` block:
```

File: website/content/docs/templates/hcl_templates/locals.mdx [hunk 6]
```diff
@@ -51,7 +50,7 @@ locals {
   name_prefix         = "${var.name_prefix != "" ? var.name_prefix : local.default_name_prefix}"
 }
 
-# Local values can be interpolated elsewhere using the "local." prefix.
+# Local variables can be interpolated elsewhere using the "local." prefix.
```
Comment (nywilken) [inline website/content/docs/templates/hcl_templates/locals.mdx:53]: Interpolation relates more to legacy JSON templates then it does HCL. 
```suggestion
# Local variables can be referenced using the "local." prefix.
```

**Review by nywilken — COMMENTED** [review]
Overall this looks good. I left a few suggestions and a question around consistency with Terraform documentation.

File: website/content/docs/templates/hcl_templates/locals.mdx (hunk 4 above)
Comment (alanszlosek) [reply website/content/docs/templates/hcl_templates/locals.mdx:2]: @nywilken Oh, so you'd rather see me go the other direction, away from "variables" towards "values" for locals? Is Terraform consistent in the usage of "values"?

I don't really see a problem with doing that. Might lead to less confusion ... put more distance between the concepts so folks realize they're quite different in behavior.

**Review by nywilken — APPROVED** [review]
LGTM. Thanks for clarifying the documentation.

**Pull request merged**